shell/platform/android/io/flutter/app/FlutterActivityDelegate.java

```diff
@@ -272,6 +286,96 @@ private boolean loadIntent(Intent intent) {
+     * {@code windowBackground} of the parent activity for use as a launch splash view.
+     *
+     * Returns null if no {@code windowBackground} is set for the activity.
+     */
+    private View createLaunchView() {
+        final Drawable launchScreenDrawable = getLaunchScreenDrawableFromActivityTheme();
+        if (launchScreenDrawable == null) {
+            return null;
+        }
+        final View view = new View(activity);
+        view.setLayoutParams(matchParent);
+        view.setBackground(launchScreenDrawable);
+        return view;
+    }
+
+    /**
+     * Extracts a {@link Drawable} from the parent activity's {@code windowBackground}.
+     *
+     * {@code android:windowBackground} is specifically reused instead of a custom defined meta-data
+     * because the Android framework can display it fast enough when launching the app as opposed
+     * to anything defined in the Activity subclass.
+     *
+     * Returns null if no {@code windowBackground} is set for the activity.
+     */
+    private Drawable getLaunchScreenDrawableFromActivityTheme() {
+        TypedValue typedValue = new TypedValue();
+        if (!activity.getTheme().resolveAttribute(
+            android.R.attr.windowBackground,
+            typedValue,
+            true)) {;
+            return null;
+        }
+        if (typedValue.resourceId == 0) {
+            return null;
+        }
+        try {
+            return activity.getResources().getDrawable(typedValue.resourceId);
+        } catch (NotFoundException e) {
+            Log.e(TAG, "Referenced launch screen drawable resource '"
+                + LAUNCH_DRAWABLE_META_DATA_KEY + "' does not exist");
+            return null;
+        }
+    }
+
+    /**
+     * Sets the root content view of the activity.
+     *
+     * If no launch screens are defined in the user application's AndroidManifest.xml as the
+     * activity's {@code windowBackground}, then set the {@link FlutterView} as the root.
+     *
+     * Otherwise, extract the {@code windowBackground}'s {@link Drawable} onto a new launch View to
+     * put in front of the {@link FlutterView}, remove the activity's {@code windowBackground},
+     * and finally remove the launch view when the {@link FlutterView} renders its first frame.
+     */
+    private void setContentView() {
+        // No transient launch screen. Set the FlutterView as root.
+        if (launchView == null) {
+            activity.setContentView(flutterView);
+            return;
+        }
+
+        final FrameLayout layout = new FrameLayout(activity);
+        layout.setLayoutParams(matchParent);
+
+        layout.addView(flutterView);
+        layout.addView(launchView);
+
+        flutterView.addFirstFrameListener(new FlutterView.FirstFrameListener() {
+            @Override
+            public void onFirstFrame() {
+                // Views need to be unparented before adding directly to activity.
+                layout.removeAllViews();
```

Comment:
The first frame callback happens on the GPU thread. But you are making non-thread safe calls into the Android framework without rethreading.